src/constraints/battery_degradation.jl

```diff
@@ -77,14 +166,37 @@ function add_degradation(m, p; b="ElectricStorage")
     add_degradation_variables(m, p)
     constrain_degradation_variables(m, p, b=b)
 
-    @constraint(m, [d in 2:days[end]],
-        m[:SOH][d] == m[:SOH][d-1] - p.hours_per_time_step * (
-            p.s.storage.attr[b].degradation.calendar_fade_coefficient * 
-            p.s.storage.attr[b].degradation.time_exponent * 
-            m[:Eavg][d-1] * d^(p.s.storage.attr[b].degradation.time_exponent-1) + 
-            p.s.storage.attr[b].degradation.cycle_fade_coefficient * m[:EFC][d-1]
+    if !isnothing(p.s.storage.attr[b].degradation.segmented_cycle_degr_bess_type) && !isnothing(p.s.storage.attr[b].degradation.segment_energy_capacity) && strategy == "augmentation"
+
+        add_segmented_cycle_fade_variables(m, p)
+        constrain_segmented_cycle_fade_variables(m, p)
+
+        segmented_cycle_fade_coefficient = Coefficient(
+            p.s.storage.attr[b].degradation.segmented_cycle_degr_bess_type,
+            p.s.storage.attr[b].degradation.segment_energy_capacity,
+            p.s.storage.attr[b].discharge_efficiency
         )
-    )
+
+        J = length(p.s.storage.attr[b].degradation.segment_energy_capacity); # Number of segments
+        @constraint(m, [d in 2:days[end]],
+            m[:SOH][d] == m[:SOH][d-1] - p.hours_per_time_step * (
+                p.s.storage.attr[b].degradation.calendar_fade_coefficient * 
+                p.s.storage.attr[b].degradation.time_exponent * 
+                m[:Eavg][d-1] * d^(p.s.storage.attr[b].degradation.time_exponent-1) + 
+                sum(segmented_cycle_fade_coefficient[j] * m[:e_minus][d-1,j] for j in 1:J)
+            )
+        )
+    else # single cycle fade parameter augmentation strategy, no segments
+        @constraint(m, [d in 2:days[end]],
+            m[:SOH][d] == m[:SOH][d-1] - p.hours_per_time_step * (
+                p.s.storage.attr[b].degradation.calendar_fade_coefficient * 
+                p.s.storage.attr[b].degradation.time_exponent * 
+                m[:Eavg][d-1] * d^(p.s.storage.attr[b].degradation.time_exponent-1) + 
+                p.s.storage.attr[b].degradation.cycle_fade_coefficient * m[:EFC][d-1]
+            )
```

Comment:
Can the SOH constraints be generalized so that the "old" model uses this code with one segment?